source/api_cc/include/custom_op.h

```diff
@@ -0,0 +1,45 @@
+#include <vector>
+#include <string>
+#include <iostream>
+
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/public/session.h"
+#include "tensorflow/core/public/version.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/shape_inference.h"
+#include <tensorflow/core/graph/default_device.h>
+#include <tensorflow/core/graph/graph_def_builder.h>
+
+using namespace tensorflow;
+using CPUDevice = Eigen::ThreadPoolDevice;
+using GPUDevice = Eigen::GpuDevice;
+
+#ifdef HIGH_PREC
+typedef double VALUETYPE;
+typedef double ENERGYTYPE;
+#else 
+typedef float  VALUETYPE;
+typedef double ENERGYTYPE;
+#endif
+
+typedef double boxtensor_t ;
+typedef double compute_t;
```

Comment:
It is better not adding the typedefs of `boxtensor_t` and `compute_t` in `custom_op.h`, because they are only used by old style op implementation.